modules/packages/Sort.chpl

```diff
@@ -802,6 +868,307 @@ proc selectionSort(Data: [?Dom] ?eltType, comparator:?rec=defaultComparator)
     compilerError("selectionSort() requires 1-D array");
 }
 
+pragma "no doc"
+proc shellSort(Data: [?Dom] ?eltType, comparator:?rec=defaultComparator,
+               start=Dom.low, end=Dom.high)
+  where !Dom.stridable
+{
+  chpl_check_comparator(comparator, eltType);
+
+  // Based on Sedgewick's Shell Sort -- see
+  // Analysis of Shellsort and Related Algorithms 1996
+  // and see Marcin Ciura - Best Increments for the Average Case of Shellsort
+  // for the choice of these increments.
+  var start_n = start;
+  var end_n = end;
+  var n = 1 + end_n - start_n;
+  var js,h,hs:int;
+  var v,tmp:Data.eltType;
+  const incs = (701, 301, 132, 57, 23, 10, 4, 1);
+  const nincs = incs.size;
+  for k in 1..nincs {
+    h = incs[k];
+    hs = h + start_n;
+    for is in hs..end_n {
```

Comment:
How about `for is in (h+start)..end` ?

Reply:
hs is used again below so I won't do this one